tests/playwright/src/model/pages/create-kind-cluster-page.ts

```diff
@@ -0,0 +1,72 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect as playExpect, type Locator, type Page } from '@playwright/test';
+
+import { CheckboxOperations } from '../core/operations';
+import { BasePage } from './base-page';
+
+export class CreateKindClusterPage extends BasePage {
+  readonly clusterPropertiesInformation: Locator;
+  readonly clusterNameField: Locator;
+  readonly controllerCheckbox: Locator;
+  readonly clusterCreationButton: Locator;
+  readonly goBackButton: Locator;
+  readonly checkboxHolder: Locator;
+  readonly logsButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.clusterPropertiesInformation = this.page.getByRole('form', { name: 'Properties Information' });
+    this.clusterNameField = this.clusterPropertiesInformation.getByRole('textbox', { name: 'Name', exact: true });
+    this.controllerCheckbox = this.clusterPropertiesInformation.getByRole('checkbox', {
+      name: 'Setup an ingress controller',
+    });
+    // Locator for the parent element of the ingress controller checkbox, used to change its value
+    this.checkboxHolder = this.controllerCheckbox.locator('..');
+    this.clusterCreationButton = this.clusterPropertiesInformation.getByRole('button', { name: 'Create', exact: true });
+    this.goBackButton = this.page.getByRole('button', { name: 'Go back to resources' });
+    this.logsButton = this.page.getByRole('button', { name: 'Show Logs' });
+  }
+
+  public async createCluster(clusterName: string): Promise<void> {
+    await playExpect(this.clusterNameField).toBeVisible();
+    if ((await this.clusterNameField.textContent()) !== clusterName) {
+      await this.clusterNameField.fill('');
+      await this.clusterNameField.fill(clusterName);
+    }
+    await playExpect(this.clusterCreationButton).toBeVisible();
+    await this.clusterCreationButton.click();
+    await playExpect(this.goBackButton).toBeVisible({ timeout: 120000 });
+    await this.goBackButton.click();
+  }
+
+  public async manageCheckboxState(operation: CheckboxOperations): Promise<void> {
+    await playExpect(this.controllerCheckbox).toBeVisible();
+    switch (operation) {
+      case CheckboxOperations.Check:
+        await this.checkboxHolder.check();
+        await playExpect(this.controllerCheckbox).toBeChecked();
+        break;
+      case CheckboxOperations.Uncheck:
+        await this.checkboxHolder.uncheck();
+        await playExpect(this.controllerCheckbox).not.toBeChecked();
+        break;
```

Comment:
You should check that the desired end state of the checkbox is not also the initial state of it, because otherwise checking and unchecking will not work as you want them. If you want to check the box you should validate that the box isnt iniitally checked before performing the check action, if it is checked initially you can just return out of the method as completed.

Reply:
The first property of the check() method is: _Ensure that element is a checkbox or a radio input. If not, this method throws. If the element is already checked, this method returns immediately_. Therefore, I believe there is no need to assert whether the element is checked before calling this method.

Reply:
Same for uncheck()

Reply:
That is not what I said, I said that you should check the checkbox is not already checked/unchecked before check/unchecking yourself.

Reply:
I would divide the create kind cluster function into two. One the will create and verifu default values in the form, and custom one, which takes multiple parameters and set them into the form.